monitoring/health_monitor.py

```
@@ -0,0 +1,341 @@
+                        pass
+            else:
+                result['status'] = 'unhealthy'
+                result['error'] = f"Unexpected status code: {response.status}"
+
+    except HTTPError as e:
+        result['status'] = 'unhealthy'
+        result['error'] = f"HTTP {e.code}"
+    except URLError as e:
+        result['status'] = 'down'
+        result['error'] = f"Connection failed: {e.reason}"
+    except socket.timeout:
```

Ruff (0.15.5)

[warning] Replace aliased errors with `TimeoutError`

Replace `socket.timeout` with builtin `TimeoutError`

(UP041)